Receiver/Sources/Receiver.swift

```diff
@@ -174,24 +174,71 @@ extension Receiver {
     }
 }
 
-extension Receiver {
-    /// Used to remove the handler from being called again (aka dispose of),
-    /// when you invoke `receiver.listen(handler)`.
-    /// Example:
-    ///
-    /// Your Receiver is shared across multiple screens (or entities) and one
-    /// of those, stops existing. In this scenario,
-    /// you would call `diposable.dispose()` at `deinit` time.
-    public class Disposable {
-        private let cleanUp: () -> Void
-
-        internal init(_ cleanUp: @escaping () -> Void) {
-            self.cleanUp = cleanUp
-        }
+/// Used to remove the handler from being called again (aka dispose of),
+/// when you invoke `receiver.listen(handler)`.
+/// Example:
+///
+/// Your Receiver is shared across multiple screens (or entities) and one
+/// of those, stops existing. In this scenario,
+/// you would call `diposable.dispose()` at `deinit` time.
+public class Disposable {
+    private let cleanUp: () -> Void
+    
+    internal init(_ cleanUp: @escaping () -> Void) {
+        self.cleanUp = cleanUp
+    }
+    
+    /// Used to dispose of the handler passed to the receiver.
+    public func dispose() {
+        cleanUp()
+    }
+    
+    /// Used to add disposable to the bag to be disposed of later
+    public func disposed(by bag: DisposeBag) {
+        bag.insert(self)
+    }
+}
 
-        /// Used to dispose of the handler passed to the receiver.
-        public func dispose() {
-            cleanUp()
+/// Used to keep a list of Disposable to dispose of them later
+/// Example:
+///
+/// In a UIViewController, you have several Receivers. You create a DisposeBag.
+/// When the UIViewController is deinit, the DisposeBag is deinit too. When
+/// the DisposeBag is deinit, all disposables will be disposed of.
+/// ```
+///     private let disposeBag = DisposeBag()
+///
+///     override func viewDidLoad() {
+///         super.viewDidLoad()
+///
+///         receiver.listen { value in
+///         //// Do whatever
+///         }.disposed(by: disposeBag)
+///     }
+/// ```
+/// No need to keep a reference to the Disposable. No need to call dispose() on
+/// the disposable. It is disposed by the DisposeBag.
+/// You can add multiple disposable to the DisposeBag.
+public class DisposeBag {
+    /// Keep a reference to all Disposable instances -> this is the actual bag
+    private var disposables = Atomic<[Disposable]>([])
```

Comment:
`disposables` should be a constant (`let`), but it's content can still change, so no need for `var`.

Reply:
You are right. Also, I was wondering if the Disposable and DisposeBag shouldn't go in a separate Disposable.swift file. What do you think?

Reply:
Makes sense, although I would prefer for them to be an extension of Receiver, since they don't make sense to be used in any other context.

Reply:
I though about that but the issue is that if you want to use the DisposeBag or the Disposable, you have to declare them like this:
`var disposeBag: Receiver<Int>.DisposeBag? = Receiver<Int>.DisposeBag()`
instead of:
`var disposeBag: DisposeBag? = DisposeBag()`

It's a lot of additional typing and it makes the Bag strongly typed.

Do you see a solution where Disposable and DisposeBag would be an extension of Receiver but where the usage of them would be like if they weren't?

Reply:
You make a good point: from a consumer point of view is a PITA, for no reason. 🤔  

Let's keep that out of the extension then. 👍